rtcp-packet.go

```diff
@@ -0,0 +1,360 @@
+  sp := new(SRTCPPacket)
+
+  // TODO: Support more than one packet
+  rtcpPacket := p.packets[0]
+
+  sp.header.buffer = make([]byte, RTCPHeaderSize)
+  copy(sp.header.buffer, rtcpPacket.header.buffer)
+
+  sp.payload = make([]byte, len(rtcpPacket.payload))
+  copy(sp.payload, rtcpPacket.payload)
+
+  block, err := aes.NewCipher(key)
+  if err != nil {
+    return nil, err
+  }
+
+  gcm, err := cipher.NewGCM(block)
+  if err != nil {
+    return nil, err
+  }
+
+  iv := p.GetGCMIV(salt)
+
+  tag := make([]byte, gcm.Overhead())
+  sp.payload = append(sp.payload, tag...)
+
+  aad := sp.header.buffer
+  aad = append(aad, p.GetESRTCPWord(true)...)
+
+  pt := sp.payload[:len(sp.payload)-len(tag)]
+
+  gcm.Seal(sp.payload[0:0], iv, pt, aad)
+
+  sp.payload = append(sp.payload, p.GetESRTCPWord(true)...)
+
+  return sp, nil
+}
+
+// https://tools.ietf.org/html/rfc7714#section-9.2
+/*
+    0                   1                   2                   3
+   0 1 2 3 4 5 6 7 8 9 0 1 2 3 4 5 6 7 8 9 0 1 2 3 4 5 6 7 8 9 0 1
+   +-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
+A  |V=2|P|   RC    |  Packet Type  |            length             |
+   +-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
+A  |           synchronization source (SSRC) of sender             |
+   +-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
+P  |                         sender info                           :
+   +-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
+P  |                        report block 1                         :
+   +-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
+P  |                        report block 2                         :
+   +-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
+P  |                              ...                              :
+   +-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
+P  |V=2|P|   SC    |  Packet Type  |              length           |
+   +=+=+=+=+=+=+=+=+=+=+=+=+=+=+=+=+=+=+=+=+=+=+=+=+=+=+=+=+=+=+=+=+
+P  |                          SSRC/CSRC_1                          |
+   +-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
+P  |                           SDES items                          :
+   +=+=+=+=+=+=+=+=+=+=+=+=+=+=+=+=+=+=+=+=+=+=+=+=+=+=+=+=+=+=+=+=+
+P  |                              ...                              :
+   +=+=+=+=+=+=+=+=+=+=+=+=+=+=+=+=+=+=+=+=+=+=+=+=+=+=+=+=+=+=+=+=+
+A  |1|                         SRTCP index                         |
+   +-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
+R  |                  SRTCP MKI (optional) index                   :
+   +-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
+R  :           SRTCP authentication tag (NOT RECOMMENDED)          :
+   +-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
+*/
+
+type SRTCPPacket struct {
+  header RTCPHeader
+  payload []byte
+}
```

Comment:
I would prefer if you followed the architecture that `RTPPacket` uses: Have a single struct `RTCPPacket` for RTCP and SRTCP, have it store a single buffer, and have accessors / mutators operate on that buffer.